Add bit about going unversioned->versioned (#2253)
temporalio/documentation: docs-src/concepts/what-is-worker-versioning.md

Unversioned Workers refer to Workers that have not opted into the Worker Versioning feature in their configuration.
They receive tasks only from Task Queues that do not have any version sets defined on them, or that have open workflows that began executing before versions were added to the queue.

To migrate from an unversioned Task Queue, add a new default Build ID to the Task Queue.
From there, deploy Workers with the same Build ID.

> Open Workflows will continue to be processed by unversioned Workers; new Workflow Executions will be process by Workers with the new Build ID.

Reviewer: ```suggestion
Unversioned Workers will continue processing open Workflows, while Workers with the new Build ID will process new Workflow Executions.
```

> Migrating from an unversioned Task Queue (ie: A Task Queue only using unversioned workers) is as simple as adding a new
> default Build ID to that Task Queue, and then deploying Workers with that Build ID. Any open Workflows will continue to
> be processed by unversioned Workers, and any new Workflows will be processed by Workers with the new Build ID.

Reviewer 2: ```suggestion
To migrate from an unversioned Task Queue, add a new default Build ID to the Task Queue.
From there, deploy Workers with the same Build ID.
Open Workflows will continue to be processed by unversioned Workers; new Workflow Executions will be process by Workers with the new Build ID.
```
